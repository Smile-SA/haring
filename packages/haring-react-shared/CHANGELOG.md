# @smile/haring-react-shared

## 0.15.0

### Minor Changes

- [#194](https://github.com/Smile-SA/haring/pull/194) [`97a610a`](https://github.com/Smile-SA/haring/commit/97a610aa02b86423b723ef8ae5d45e8ecf346f92) Thanks [@vapersmile](https://github.com/vapersmile)! - Upgrade mantine version to 7.11.x

## 0.14.0

## 0.13.0

### Minor Changes

- [#178](https://github.com/Smile-SA/haring/pull/178) [`ce8f39a2c52590b2c91950450a84222434913743`](https://github.com/Smile-SA/haring/commit/ce8f39a2c52590b2c91950450a84222434913743) Thanks [@tonai](https://github.com/tonai)! - Rename packages

## 0.12.0

## 0.11.0

## 0.10.0

### Minor Changes

- [#149](https://github.com/Smile-SA/haring/pull/149) [`2706a09`](https://github.com/Smile-SA/haring/commit/2706a097fd86b072cf21d12fe2a97427883671cf) Thanks [@vapersmile](https://github.com/vapersmile)! - Add Calendar style on globlaStyle and add a preview

- [#150](https://github.com/Smile-SA/haring/pull/150) [`34fabfb`](https://github.com/Smile-SA/haring/commit/34fabfbd1a2c048c39adc567548a3ee7e85074ee) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Fixed missing global styles for focus-visible color

## 0.9.0

### Minor Changes

- [`d430e2c`](https://github.com/Smile-SA/haring/commit/d430e2cbcd26871c4db0c53f92a4e4baa6af839c) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Migrated from Mantine V6 to V7

## 0.8.1

## 0.8.0

### Minor Changes

- [#108](https://github.com/Smile-SA/haring/pull/108) [`320634d`](https://github.com/Smile-SA/haring/commit/320634d7ec2048a328cfda1dbcff5d1b70a32563) Thanks [@vapersmile](https://github.com/vapersmile)! - Add custom default style for carousel mantine composent and add 2-Mantine/carousel page

## 0.7.0

### Minor Changes

- [#87](https://github.com/Smile-SA/haring/pull/87) [`dbc06ca`](https://github.com/Smile-SA/haring/commit/dbc06ca55961b69663ab7fdc02609c6525ae361d) Thanks [@tonai](https://github.com/tonai)! - Add theme provider and update barrel file for better next.js compatibility

- [#95](https://github.com/Smile-SA/haring/pull/95) [`5696400`](https://github.com/Smile-SA/haring/commit/5696400e6f703da52db5f7199b50f8251fa76726) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `FilterList` component, renamed `FiltersBar` in `SidebarFilters`
  and `FiltersCheckboxList` in `SearchableCheckboxList`,
  refactored `SearchableCheckboxList`, added common `IFilter` type in shared
  package, added topContent prop to `SwitchableView`, fixed `BrowsingPage` with
  name changes.

## 0.6.0

### Minor Changes

- [#74](https://github.com/Smile-SA/haring/pull/74) [`0ad5377`](https://github.com/Smile-SA/haring/commit/0ad5377535e7c19941da301e8a2ee7298ab70f91) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Refactored actions in `ThumbnailGrid` and `TableGridView` components, moved `typeGuard` helper to shared package, added `typeGuardInterface` function, added tests

## 0.5.0

### Minor Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add dynamic componentProps property in actions

- [#65](https://github.com/Smile-SA/haring/pull/65) [`bd9cfe4`](https://github.com/Smile-SA/haring/commit/bd9cfe42d0b22f5f7f5e7b0de30fdfb22ad3e1c8) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Reworked action/confirm types and moved then in `react-front-kit-shared` then refactored types in `Table`, `Thumbnail` and `ConfirmModal` , Added `ThumbnailGrid` and `TableGridView` components,

- [#68](https://github.com/Smile-SA/haring/pull/68) [`c8b5a39`](https://github.com/Smile-SA/haring/commit/c8b5a3978c8ce7133da0b498d9f0b326f07eb737) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `storybook-pages` package containing example pages, renamed `TestPage`
  into `BrowsingPage` and completed it with integration of `TableGridView`, fixed
  various component styles and props related to page.

### Patch Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add server barrel files

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/haring/pull/53) [`01ee3ea`](https://github.com/Smile-SA/haring/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
