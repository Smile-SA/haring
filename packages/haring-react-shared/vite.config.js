import { resolve } from 'path';

import react from '@vitejs/plugin-react-swc';
import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.tsx'),
      fileName: '[name]',
      formats: ['cjs', 'es'],
    },
    rollupOptions: {
      external: [
        '@mantine/carousel',
        '@mantine/core',
        '@mantine/dates',
        '@mantine/dropzone',
        '@mantine/hooks',
        '@phosphor-icons/react',
        '@smile/haring-react',
        '@smile/haring-react-dates',
        '@smile/haring-react-dropzone',
        '@smile/haring-react-shared',
        '@smile/haring-react-table',
        '@storybook/addon-actions',
        '@storybook/preview-api',
        '@tabler/icons-react',
        '@testing-library/react',
        'dayjs',
        'embla-carousel-react',
        'mantine-react-table',
        'react',
        'react/jsx-runtime',
      ],
      input: {
        index: resolve(__dirname, 'src/index.tsx'),
        server: resolve(__dirname, 'src/server.tsx'),
        'storybook-utils': resolve(__dirname, 'src/storybook-utils/index.ts'),
        'test-utils': resolve(__dirname, 'src/test-utils/index.ts'),
      },
      output: {
        banner: (chunkInfo) => {
          if (
            ['src/index.tsx'].find((modulePath) =>
              chunkInfo.facadeModuleId?.endsWith(modulePath),
            )
          ) {
            return `"use client"`;
          }
          return '';
        },
        preserveModules: false,
      },
    },
  },
  plugins: [
    react(),
    dts({
      insertTypesEntry: true,
      tsconfigPath: './tsconfig.build.json',
    }),
  ],
});
