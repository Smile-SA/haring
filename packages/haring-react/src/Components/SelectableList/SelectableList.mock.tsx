import { Button, Space } from '@mantine/core';
import { DownloadSimple } from '@phosphor-icons/react';

import { DocumentBox } from '../DocumentBox/DocumentBox';

export const selectableListElementsMock = [
  <DocumentBox
    key={1}
    author="Aline Dupon"
    date="Published on December 24, 2023"
    iconType="PDF"
    path="(Customer > 567890456 > Invoices)"
    title="Random_File.PDF"
  >
    <>
      <p>
        Ceci est une description faite pour cette facture et ajoutée par le
        créateur lors de l’import du document dans la GED, en l’absence de
        description cet espace est laissé vide...
      </p>
      <Button color="gray.8">
        <DownloadSimple width={12} />
        <Space w={8} />
        PDF, FR - 1Mo
      </Button>
    </>
  </DocumentBox>,
  <DocumentBox
    key={2}
    author="Julien Dominique"
    date="Published on December 24, 2023"
    iconType="ppt"
    path="(Customer > 567890456 > Invoices)"
    title="Presentation.PPT"
  >
    <>
      <p>
        Ceci est une description faite pour cette facture et ajoutée par le
        créateur lors de l’import du document dans la GED, en l’absence de
        description cet espace est laissé vide...
      </p>
      <Button color="gray.8">
        <DownloadSimple width={12} />
        <Space w={8} />
        PTT, FR - 1Mo
      </Button>
    </>
  </DocumentBox>,
  <DocumentBox
    key={3}
    author="Mohamed Aldri"
    date="Published on December 24, 2023"
    iconType="PDF"
    path="(Customer > 567890456 > Invoices)"
    title="Other_random_File.PDF"
  >
    <>
      <p>
        Ceci est une description faite pour cette facture et ajoutée par le
        créateur lors de l’import du document dans la GED, en l’absence de
        description cet espace est laissé vide...
      </p>
      <Button color="gray.8">
        <DownloadSimple width={12} />
        <Space w={8} />
        PDF, FR - 1Mo
      </Button>
    </>
  </DocumentBox>,
];
