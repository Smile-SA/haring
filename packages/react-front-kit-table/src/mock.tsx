/* eslint-disable react-refresh/only-export-components */
export * from './Components/Table/Table.mock';
export * from './Components/TableGridView/TableGridView.mock';
