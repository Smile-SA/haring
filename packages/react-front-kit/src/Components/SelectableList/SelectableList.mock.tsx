import { Button, Space } from '@mantine/core';
import { DownloadSimple } from '@phosphor-icons/react';

import { DocumentCard } from '../DocumentCard/DocumentCard';

export const selectableListElementsMock = [
  <DocumentCard
    key={1}
    author="Aline Dupon"
    date="Published on December 24, 2023"
    iconType="PDF"
    path="(Customer > 567890456 > Invoices)"
    title="Random_File.PDF"
  >
    <>
      <p>
        Ceci est une description faite pour cette facture et ajoutée par le
        créateur lors de l’import du document dans la GED, en l’absence de
        description cet espace est laissé vide...
      </p>
      <Button color="gray.8">
        <DownloadSimple width={12} />
        <Space w={8} />
        PDF, FR - 1Mo
      </Button>
    </>
  </DocumentCard>,
  <DocumentCard
    key={2}
    author="Julien Dominique"
    date="Published on December 24, 2023"
    iconType="ppt"
    path="(Customer > 567890456 > Invoices)"
    title="Presentation.PPT"
  >
    <>
      <p>
        Ceci est une description faite pour cette facture et ajoutée par le
        créateur lors de l’import du document dans la GED, en l’absence de
        description cet espace est laissé vide...
      </p>
      <Button color="gray.8">
        <DownloadSimple width={12} />
        <Space w={8} />
        PTT, FR - 1Mo
      </Button>
    </>
  </DocumentCard>,
  <DocumentCard
    key={3}
    author="Mohamed Aldri"
    date="Published on December 24, 2023"
    iconType="PDF"
    path="(Customer > 567890456 > Invoices)"
    title="Other_random_File.PDF"
  >
    <>
      <p>
        Ceci est une description faite pour cette facture et ajoutée par le
        créateur lors de l’import du document dans la GED, en l’absence de
        description cet espace est laissé vide...
      </p>
      <Button color="gray.8">
        <DownloadSimple width={12} />
        <Space w={8} />
        PDF, FR - 1Mo
      </Button>
    </>
  </DocumentCard>,
];
