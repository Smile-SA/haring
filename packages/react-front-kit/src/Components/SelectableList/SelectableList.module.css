.element {
  &:not(:last-child):after {
    /*noinspection CssInvalidAtRule*/
    @mixin smaller-than $mantine-breakpoint-sm {
      bottom: -25px;
    }
    border: 1px #e9ecef solid;
    bottom: -41px;
    content: '';
    position: absolute;
    width: 100%;
  }

  position: relative;
}

.list {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    gap: 48px;
  }
  gap: 80px;
}
