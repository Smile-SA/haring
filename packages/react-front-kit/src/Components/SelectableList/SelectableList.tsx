import type { StackProps } from '@mantine/core';
import type { ReactElement } from 'react';

import { Checkbox, Group, Stack } from '@mantine/core';
import { createStyles } from '@mantine/styles';

const useStyles = createStyles(() => ({
  element: {
    ':not(:last-child):after': {
      border: '1px #e9ecef solid',
      bottom: '-41px',
      content: '""',
      position: 'absolute',
      width: '100%',
    },
    position: 'relative',
  },
}));

export interface ISelectableListProps extends StackProps {
  children: ReactElement[];
  onSelectChange?: (index: number, isSelected: boolean) => void;
  selectedIndexes?: number[];
}

export function SelectableList(props: ISelectableListProps): ReactElement {
  const {
    children,
    onSelectChange,
    selectedIndexes = [],
    ...stackProps
  } = props;

  const { classes } = useStyles();

  return (
    <Stack spacing={80} {...stackProps}>
      {children.map((element, index) => (
        // eslint-disable-next-line react/no-array-index-key
        <Group
          key={element.key ?? index}
          className={classes.element}
          noWrap
          spacing={32}
        >
          <Checkbox
            checked={selectedIndexes.includes(index)}
            onChange={(e) => onSelectChange?.(index, e.currentTarget.checked)}
            radius={2}
            size={16}
          />
          {element}
        </Group>
      ))}
    </Stack>
  );
}
