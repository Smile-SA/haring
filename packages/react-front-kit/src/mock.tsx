/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
