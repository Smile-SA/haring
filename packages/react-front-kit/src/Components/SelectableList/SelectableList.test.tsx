import { renderWithProviders } from '@smile/haring-react-shared/test-utils';

import { SelectableList } from './SelectableList';
import { selectableListElementsMock } from './SelectableList.mock';

describe('SelectableList', () => {
  beforeEach(() => {
    // Prevent mantine random ID
    Math.random = () => 0.42;
  });
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <SelectableList selectedIndexes={[1]}>
        {selectableListElementsMock}
      </SelectableList>,
    );
    expect(container).toMatchSnapshot();
  });
});
