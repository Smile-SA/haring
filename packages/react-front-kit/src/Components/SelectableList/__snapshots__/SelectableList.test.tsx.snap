// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectableList matches snapshot 1`] = `
<div>
  <div
    class="mantine-Stack-root mantine-uyiazu"
  >
    <div
      class="mantine-Group-root mantine-1bxm023"
    >
      <div
        class="mantine-Checkbox-root mantine-yxmaw9"
      >
        <div
          class="mantine-hqx0d1 mantine-Checkbox-body"
        >
          <div
            class="mantine-13uu19v mantine-Checkbox-inner"
          >
            <input
              class="mantine-1g8epsg mantine-Checkbox-input"
              id="mantine-f4bipx4bi"
              type="checkbox"
            />
            <svg
              class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
              fill="none"
              viewBox="0 0 10 7"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="mantine-djumwf mantine-Checkbox-labelWrapper"
          />
        </div>
      </div>
      <div
        class="mantine-1feahhc"
      >
        <div
          class="mantine-Image-root mantine-1hgbf0p"
          style="width: 100%;"
        >
          <figure
            class="mantine-qenwvq mantine-Image-figure"
          >
            <div
              class="mantine-1iugybl mantine-Image-imageWrapper"
            >
              <img
                class="mantine-Image-image mantine-1thijn1"
                src=""
                style="object-fit: cover; width: 100%; height: auto;"
              />
            </div>
          </figure>
        </div>
        <div>
          <div>
            <div
              class="mantine-mdzvt7"
            >
              <div
                class="mantine-qu0qb1"
              >
                <svg
                  fill="#0b7285"
                  height="38"
                  viewBox="0 0 256 256"
                  width="38"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M222,152a6,6,0,0,1-6,6H190v20h18a6,6,0,0,1,0,12H190v18a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6h32A6,6,0,0,1,222,152ZM90,172a26,26,0,0,1-26,26H54v10a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6H64A26,26,0,0,1,90,172Zm-12,0a14,14,0,0,0-14-14H54v28H64A14,14,0,0,0,78,172Zm84,8a34,34,0,0,1-34,34H112a6,6,0,0,1-6-6V152a6,6,0,0,1,6-6h16A34,34,0,0,1,162,180Zm-12,0a22,22,0,0,0-22-22H118v44h10A22,22,0,0,0,150,180ZM42,112V40A14,14,0,0,1,56,26h96a6,6,0,0,1,4.25,1.76l56,56A6,6,0,0,1,214,88v24a6,6,0,0,1-12,0V94H152a6,6,0,0,1-6-6V38H56a2,2,0,0,0-2,2v72a6,6,0,0,1-12,0ZM158,82h35.52L158,46.48Z"
                  />
                </svg>
              </div>
              <div>
                <div
                  class="mantine-dxs7ne"
                >
                  <span
                    class="mantine-fakv65"
                  >
                    Random_File.PDF
                  </span>
                  <span
                    class="mantine-xrrpki"
                  >
                    (Customer &gt; 567890456 &gt; Invoices)
                  </span>
                </div>
                <div
                  class="mantine-1fna5yi"
                >
                  <span>
                    Published on December 24, 2023
                  </span>
                  <span
                    class="mantine-mj4lz4"
                  >
                    -
                  </span>
                  <span
                    class="mantine-l9up18"
                  >
                    Aline Dupon
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mantine-j5huwo"
          >
            <p>
              Ceci est une description faite pour cette facture et ajoutée par le créateur lors de l’import du document dans la GED, en l’absence de description cet espace est laissé vide...
            </p>
            <button
              class="mantine-UnstyledButton-root mantine-Button-root mantine-1tate3y"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  class="mantine-1ryt1ht mantine-Button-label"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 256 256"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224,152v56a16,16,0,0,1-16,16H48a16,16,0,0,1-16-16V152a8,8,0,0,1,16,0v56H208V152a8,8,0,0,1,16,0Zm-101.66,5.66a8,8,0,0,0,11.32,0l40-40a8,8,0,0,0-11.32-11.32L136,132.69V40a8,8,0,0,0-16,0v92.69L93.66,106.34a8,8,0,0,0-11.32,11.32Z"
                    />
                  </svg>
                  <div
                    class="mantine-151bw52"
                  />
                  PDF, FR - 1Mo
                </span>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mantine-Group-root mantine-1bxm023"
    >
      <div
        class="mantine-Checkbox-root mantine-yxmaw9"
      >
        <div
          class="mantine-hqx0d1 mantine-Checkbox-body"
        >
          <div
            class="mantine-13uu19v mantine-Checkbox-inner"
          >
            <input
              checked=""
              class="mantine-1g8epsg mantine-Checkbox-input"
              id="mantine-f4bipx4bi"
              type="checkbox"
            />
            <svg
              class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
              fill="none"
              viewBox="0 0 10 7"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="mantine-djumwf mantine-Checkbox-labelWrapper"
          />
        </div>
      </div>
      <div
        class="mantine-1feahhc"
      >
        <div
          class="mantine-Image-root mantine-1hgbf0p"
          style="width: 100%;"
        >
          <figure
            class="mantine-qenwvq mantine-Image-figure"
          >
            <div
              class="mantine-1iugybl mantine-Image-imageWrapper"
            >
              <img
                class="mantine-Image-image mantine-1thijn1"
                src=""
                style="object-fit: cover; width: 100%; height: auto;"
              />
            </div>
          </figure>
        </div>
        <div>
          <div>
            <div
              class="mantine-mdzvt7"
            >
              <div
                class="mantine-qu0qb1"
              >
                <svg
                  fill="#0b7285"
                  height="38"
                  viewBox="0 0 256 256"
                  width="38"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M222,152a6,6,0,0,1-6,6H202v50a6,6,0,0,1-12,0V158H176a6,6,0,0,1,0-12h40A6,6,0,0,1,222,152ZM90,172a26,26,0,0,1-26,26H54v10a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6H64A26,26,0,0,1,90,172Zm-12,0a14,14,0,0,0-14-14H54v28H64A14,14,0,0,0,78,172Zm80,0a26,26,0,0,1-26,26H122v10a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6h16A26,26,0,0,1,158,172Zm-12,0a14,14,0,0,0-14-14H122v28h10A14,14,0,0,0,146,172ZM42,112V40A14,14,0,0,1,56,26h96a6,6,0,0,1,4.25,1.76l56,56A6,6,0,0,1,214,88v24a6,6,0,0,1-12,0V94H152a6,6,0,0,1-6-6V38H56a2,2,0,0,0-2,2v72a6,6,0,0,1-12,0ZM158,82h35.52L158,46.48Z"
                  />
                </svg>
              </div>
              <div>
                <div
                  class="mantine-dxs7ne"
                >
                  <span
                    class="mantine-fakv65"
                  >
                    Presentation.PPT
                  </span>
                  <span
                    class="mantine-xrrpki"
                  >
                    (Customer &gt; 567890456 &gt; Invoices)
                  </span>
                </div>
                <div
                  class="mantine-1fna5yi"
                >
                  <span>
                    Published on December 24, 2023
                  </span>
                  <span
                    class="mantine-mj4lz4"
                  >
                    -
                  </span>
                  <span
                    class="mantine-l9up18"
                  >
                    Julien Dominique
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mantine-j5huwo"
          >
            <p>
              Ceci est une description faite pour cette facture et ajoutée par le créateur lors de l’import du document dans la GED, en l’absence de description cet espace est laissé vide...
            </p>
            <button
              class="mantine-UnstyledButton-root mantine-Button-root mantine-1tate3y"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  class="mantine-1ryt1ht mantine-Button-label"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 256 256"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224,152v56a16,16,0,0,1-16,16H48a16,16,0,0,1-16-16V152a8,8,0,0,1,16,0v56H208V152a8,8,0,0,1,16,0Zm-101.66,5.66a8,8,0,0,0,11.32,0l40-40a8,8,0,0,0-11.32-11.32L136,132.69V40a8,8,0,0,0-16,0v92.69L93.66,106.34a8,8,0,0,0-11.32,11.32Z"
                    />
                  </svg>
                  <div
                    class="mantine-151bw52"
                  />
                  PTT, FR - 1Mo
                </span>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mantine-Group-root mantine-1bxm023"
    >
      <div
        class="mantine-Checkbox-root mantine-yxmaw9"
      >
        <div
          class="mantine-hqx0d1 mantine-Checkbox-body"
        >
          <div
            class="mantine-13uu19v mantine-Checkbox-inner"
          >
            <input
              class="mantine-1g8epsg mantine-Checkbox-input"
              id="mantine-f4bipx4bi"
              type="checkbox"
            />
            <svg
              class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
              fill="none"
              viewBox="0 0 10 7"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="mantine-djumwf mantine-Checkbox-labelWrapper"
          />
        </div>
      </div>
      <div
        class="mantine-1feahhc"
      >
        <div
          class="mantine-Image-root mantine-1hgbf0p"
          style="width: 100%;"
        >
          <figure
            class="mantine-qenwvq mantine-Image-figure"
          >
            <div
              class="mantine-1iugybl mantine-Image-imageWrapper"
            >
              <img
                class="mantine-Image-image mantine-1thijn1"
                src=""
                style="object-fit: cover; width: 100%; height: auto;"
              />
            </div>
          </figure>
        </div>
        <div>
          <div>
            <div
              class="mantine-mdzvt7"
            >
              <div
                class="mantine-qu0qb1"
              >
                <svg
                  fill="#0b7285"
                  height="38"
                  viewBox="0 0 256 256"
                  width="38"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M222,152a6,6,0,0,1-6,6H190v20h18a6,6,0,0,1,0,12H190v18a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6h32A6,6,0,0,1,222,152ZM90,172a26,26,0,0,1-26,26H54v10a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6H64A26,26,0,0,1,90,172Zm-12,0a14,14,0,0,0-14-14H54v28H64A14,14,0,0,0,78,172Zm84,8a34,34,0,0,1-34,34H112a6,6,0,0,1-6-6V152a6,6,0,0,1,6-6h16A34,34,0,0,1,162,180Zm-12,0a22,22,0,0,0-22-22H118v44h10A22,22,0,0,0,150,180ZM42,112V40A14,14,0,0,1,56,26h96a6,6,0,0,1,4.25,1.76l56,56A6,6,0,0,1,214,88v24a6,6,0,0,1-12,0V94H152a6,6,0,0,1-6-6V38H56a2,2,0,0,0-2,2v72a6,6,0,0,1-12,0ZM158,82h35.52L158,46.48Z"
                  />
                </svg>
              </div>
              <div>
                <div
                  class="mantine-dxs7ne"
                >
                  <span
                    class="mantine-fakv65"
                  >
                    Other_random_File.PDF
                  </span>
                  <span
                    class="mantine-xrrpki"
                  >
                    (Customer &gt; 567890456 &gt; Invoices)
                  </span>
                </div>
                <div
                  class="mantine-1fna5yi"
                >
                  <span>
                    Published on December 24, 2023
                  </span>
                  <span
                    class="mantine-mj4lz4"
                  >
                    -
                  </span>
                  <span
                    class="mantine-l9up18"
                  >
                    Mohamed Aldri
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mantine-j5huwo"
          >
            <p>
              Ceci est une description faite pour cette facture et ajoutée par le créateur lors de l’import du document dans la GED, en l’absence de description cet espace est laissé vide...
            </p>
            <button
              class="mantine-UnstyledButton-root mantine-Button-root mantine-1tate3y"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  class="mantine-1ryt1ht mantine-Button-label"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 256 256"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224,152v56a16,16,0,0,1-16,16H48a16,16,0,0,1-16-16V152a8,8,0,0,1,16,0v56H208V152a8,8,0,0,1,16,0Zm-101.66,5.66a8,8,0,0,0,11.32,0l40-40a8,8,0,0,0-11.32-11.32L136,132.69V40a8,8,0,0,0-16,0v92.69L93.66,106.34a8,8,0,0,0-11.32,11.32Z"
                    />
                  </svg>
                  <div
                    class="mantine-151bw52"
                  />
                  PDF, FR - 1Mo
                </span>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
